### DISCUSSION TITLE
ALSA MIXER - no device found

### DISCUSSION THREAD

**Discussion by kw-sudo** [rern/rAudio#1066] [Bug Reports]
Since a few days r/Audio sometimes didn't find my output device (USB A-B) and wasn't shown in config menu...

![Bildschirmfoto 2022-10-13 um 18 02 56](https://user-images.githubusercontent.com/104136223/195647643-f980e930-31a6-4a38-855f-72b559a143d3.jpg)

ssh--> alsamixer

`ALSA lib confmisc.c:855:(parse_card) cannot find card '-1'
ALSA lib conf.c:5180:(_snd_config_evaluate) function snd_func_card_inum returned error: Kein passendes Gerät gefunden
ALSA lib confmisc.c:422:(snd_func_concat) error evaluating strings
ALSA lib conf.c:5180:(_snd_config_evaluate) function snd_func_concat returned error: Kein passendes Gerät gefunden
ALSA lib confmisc.c:1334:(snd_func_refer) error evaluating name
ALSA lib conf.c:5180:(_snd_config_evaluate) function snd_func_refer returned error: Kein passendes Gerät gefunden
ALSA lib conf.c:5703:(snd_config_expand) Evaluate error: Kein passendes Gerät gefunden
ALSA lib control.c:1528:(snd_ctl_open_noupdate) Invalid CTL default
Fehler beim Öffen des Mixer-Gerätes: Kein passendes Gerät gefunden`

**OR**

is present but no sound. 

Therefore I have to un-/plug USB-cable or switch output device from onboard-headphone and back to amp. Can I disable the onboard-headphone permanently? (asound.conf)

PS: my amp has an auto-power function. Shutdown/reboot r/Audio, amp --> power off, too. r/Audio start --> power on, so there's still an USB signal.

**Comment by kw-sudo** [comment]
Issue "solved"...new installation, new luck. ;-)

**Comment by kw-sudo** [comment]
The issue is still going on. How can I disable the onboard sound/headphones permanently so r/Audio can only set USB to active?

**Comment by feebb** [comment]
You might try commenting out dtparam=audio=on in /boot/config.txt.

`#dtparam=audio=on`

**Reply by kw-sudo** [reply]
feebb, you made my day..that's it! 😀 After a reboot there's no more dropdown menu in player settings and only my amp available.

**Reply by kw-sudo** [reply]
Happy too soon... The amplifier is the only device shown, but when I switch the signal source from USB to HDMI (eARC TV) and back on the amplifier, r/Audio loses the signal. Even a restart doesn't help. I either have to pull the cable or briefly change the sound source on the Raspberry (onboard <--> amp)

**Comment by kw-sudo** [comment]
I can't confirm it 100%, but the error doesn't occur as long as there are songs in the playlist. If I clear the list manually and change the source on the amplifier, the problem occurs

**Reply by kw-sudo** [reply]
The last written is only part of the solution, the USB input must be active before starting r/audio. Starting before, e.g. on HDMI, the r/audio sound output freezes.

**Reply by kw-sudo** [reply]
Still unclear what causes the issue..
